### PR TITLE
fix(iteration): fix user experience for iteration block and icon

### PR DESCRIPTION
**Pull request by SMahil**
### [ ] What does this PR do? _[Mandatory]_
          Fix User Experience for Iteration block

### [ ] What issue/task does this PR references? _[Mandatory]_
https://github.com/openshiftio/openshift.io/issues/2652

### [ ] Are the tests Included? _[Mandatory]_
NA

#### [ ] Is the documentation Included?
NA

#### [ ] Are the Release Notes included?
NA

#### [ ] Is/Are the @mention(s) to reviewer(s) included
@nimishamukherjee @rgarg1

### CODE REVIEW

**Comment by debloper** [comment]
Subset patch of #2564

**Comment by nimishamukherjee** [comment]
[test]

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2529-10 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2529-fabric8-planner.badger.fabric8.io

File: src/app/components_ngrx/iteration-list-entry/iteration-list-entry.component.less [hunk 1]
```diff
@@ -14,6 +14,13 @@
       width: 15px;
       padding: 5px;
       font-size: 16px;
+      &:hover {
+        color: @color-pf-blue;
+        cursor: pointer;
```
Comment (kwk) [inline src/app/components_ngrx/iteration-list-entry/iteration-list-entry.component.less]: @SMahil @debloper why do we have to be so detailed about everything? Shouldn't a link automatically get a pointy cursor?

Comment (SMahil) [reply src/app/components_ngrx/iteration-list-entry/iteration-list-entry.component.less]: @kwk It should be but right now it's not the case. I can plan a thorough check to fix this kind of issues.

Comment (debloper) [reply src/app/components_ngrx/iteration-list-entry/iteration-list-entry.component.less]: @kwk yes, it automatically becomes pointer (unless forced not to) if we're talking about the same anchor element. However, here the style is getting applied on a div element, and the anchor is a different child element. Due to classic over-nesting of DOM in bootstrap framework that we tend to get used to, and carry it around - situations like this happen (and is often overlooked).

@SMahil I've verified this in my PR (where the DOM is restructured to a leaner functional alternative with less container nesting), if you set the main clickable object itself as anchor tag (the way it should be), then it won't be necessary to explicitly set cursors with CSS.

Comment (SMahil) [reply src/app/components_ngrx/iteration-list-entry/iteration-list-entry.component.less]: @kwk @debloper this is for icon not for the iteration name block

**Comment by SMahil** [comment]
Closing this PR as issue is being tracked at https://github.com/fabric8-ui/fabric8-planner/pull/2564

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2529-14 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2529-fabric8-planner.badger.fabric8.io

**Pull request closed without merging**